core/src/main/java/eu/fasten/core/data/metadatadb/MetadataDao.java

```diff
@@ -330,6 +331,43 @@ public long insertModule(long packageVersionId, String namespace,
         return resultRecord.getValue(Modules.MODULES.ID);
     }
 
+    /**
+     * Inserts a record in 'modules' table in the database.
+     *
+     * @param packageVersionId ID of the package version where the module belongs
+     *                         (references 'package_versions.id')
+     * @param namespace        Namespace of the module
+     * @param createdAt        Timestamp when the module was created
+     * @param metadata         Metadata of the module
+     * @return ID of the new record
+     */
+    public long insertModule(long packageVersionId, String namespace,
+                             Timestamp createdAt, JSONObject metadata,
+                             boolean addDuplicates) {
+        var metadataJsonb = metadata != null ? JSONB.valueOf(metadata.toString()) : null;
+        var resultRecord = context.insertInto(Modules.MODULES,
+                Modules.MODULES.PACKAGE_VERSION_ID, Modules.MODULES.NAMESPACE,
+                Modules.MODULES.CREATED_AT, Modules.MODULES.METADATA)
+                .values(packageVersionId, namespace, createdAt, metadataJsonb)
+                // FIXME
```

Comment:
What is the problem here?

Reply:
This was added on #91 . @StefanosChaliasos could you please take a look?